### PR TITLE
Wagtail 4 compatibility

### PR DESCRIPTION
**Pull request by nickmoreton**
This fixes testing for Wagtail v4 and updates the documentation to be inline with Wagtail 3+

I didn't need to alter any of the core package files.

### CODE REVIEW

**Review by katdom13 — DISMISSED** [review]
Looks good to me

**Comment by mdfathi99** [comment]
Will it be released soon?

**Review by mdfathi99 — DISMISSED** [review]
As per wagtail 4.0 docs this commit is correct.

**Review by mdfathi99 — COMMENTED** [review]
As per wagtail 4.0 everything seems correct.

File: docs/basic_usage.rst [hunk 1]
```diff
@@ -54,12 +54,12 @@ AutocompletePanel
...
 
     .. note::
-        Unlike :class:`~wagtail:wagtail.wagtailadmin.edit_handlers.PageChooserPanel`,
+        Unlike :class:`~wagtail:wagtail.admin.panels.FieldPanel`,
         ``AutocompletePanel`` does not support receiving ``target_model`` as a list.
 
     .. note::
         ``AutocompletePanel`` does not support receiving the ``can_choose_root``
-        argument that :class:`~wagtail:wagtail.wagtailadmin.edit_handlers.PageChooserPanel`
+        argument that :class:`~wagtail:wagtail.admin.panels.FieldPanel`
```
Comment (bcdickinson) [inline docs/basic_usage.rst:62]: You still need to use `PageChooserPanel` if you pass the `can_choose_root` kwarg

File: docs/basic_usage.rst [hunk 2]
```diff
@@ -21,13 +21,13 @@ We have a ``BlogPage`` that lets the editor select an ``AuthorPage`` page.
...
 The ``AuthorPage`` would traditionally be selected with a
-:class:`~wagtail:wagtail.wagtailadmin.edit_handlers.PageChooserPanel`,
+:class:`~wagtail:wagtail.admin.panels.FieldPanel`,
 like the following.
 
 .. code-block:: python
 
     content_panels = Page.content_panels + [
-        PageChooserPanel('author', page_type='app_label.AuthorPage'),
+        FieldPanel('author', page_type='app_label.AuthorPage'),
```
[the remark this answers is not in the document]
Comment (nickmoreton) [reply docs/basic_usage.rst]: Thanks @bcdickinson This should be right now.

File: docs/basic_usage.rst (hunk 1 above)
Comment (nickmoreton) [reply docs/basic_usage.rst]: This should be right as well but I thought it could do with a little more clarity injected.

File: tox.ini [hunk 3]
```diff
@@ -2,8 +2,9 @@
 skipsdist = True
 usedevelop = True
 envlist =
-    py{37,38,39}-dj{30,31,32}-wt{215}
-    py{37,38,39}-dj{32,40}-wt{216,30}
+    py{37,38,39,310}-dj{32}-wt{215,216}
+    py{38,39,310}-dj{32,40}-wt{216,30,40}
+    py{38,39,310,311}-dj{41}-wt{40}
```
Comment (nickmoreton) [inline tox.ini:7]: When adding Wagtail 4.1 the tests are failing but the CMS interface seems to be working OK.
I can look at this when we need it for Wagtail 4.1

File: tox.ini [hunk 4]
```diff
@@ -2,8 +2,9 @@
 skipsdist = True
 usedevelop = True
 envlist =
-    py{37,38,39}-dj{30,31,32}-wt{215}
-    py{37,38,39}-dj{32,40}-wt{216,30}
+    py{37,38,39}-dj{32}-wt{215,216}
+    py{38,39}-dj{32,40}-wt{216,30,40}
+    py{38,39}-dj{41}-wt{40}
```
Comment (nickmoreton) [inline tox.ini:7]: **Questions**: How can python 3.10 and 3.11 be run in Circle CI?

**Comment by harrislapiroff** [comment]
Thanks for the PR @nickmoreton—we'll take a look soon.

**Comment by harrislapiroff** [comment]
I mentioned this in #143 but as you've observed, tox fails when adding newer versions of Python. It looks like the [Docker image we're using in CI](https://github.com/wagtail/wagtail-autocomplete/blob/main/.circleci/config.yml#L6) doesn't have those versions. That image may be unmaintained. We'll need to find a new image/write our own. Is that something you can do? Otherwise we can take a stab at it on our end, but no promises about timing.

**Comment by nickmoreton** [comment]
I'm closing this PR as I don't believe it required. That package works OK on wagtail up to v5 when I use it on an upgrade site.

**Pull request closed without merging**